logstash-core/src/main/java/org/logstash/Logstash.java

```diff
@@ -35,21 +35,39 @@ public static void main(final String... args) {
         final String lsHome = System.getenv("LS_HOME");
         if (lsHome == null) {
             throw new IllegalStateException(
-                "LS_HOME environment variable must be set. This is likely a bug that should be reported."
+                    "LS_HOME environment variable must be set. This is likely a bug that should be reported."
             );
         }
         final Path home = Paths.get(lsHome).toAbsolutePath();
         try (
-            final Logstash logstash = new Logstash(home, args, System.out, System.err, System.in)
+                final Logstash logstash = new Logstash(home, args, System.out, System.err, System.in)
         ) {
             logstash.run();
+        } catch (final IllegalStateException e) {
+            if (e.getMessage().contains("Could not load FFI Provider")) {
+                inaccessibleTempDirWarning();
+            }
+            handleCriticalError(e);
```

Comment:
This may seem like a nitpick, but could we have the stacktrace print above the tmpwarning? 

People usually read error logs bottom to top, and I think people might find that less confusing. I've gotten stuck on a stacktrace only to realize later that a much more relevant error was right above it.